ranchhand.tf

```diff
@@ -1,17 +1,20 @@
 locals {
   ranchhand_cert_ips = concat([local.lb_ip], var.ranchhand_cert_ipaddresses)
 
-  node_ips = formatlist(
-          (var.enable_public_instances ? "%s:%s" : "%[2]s"),
-          var.enable_public_instances ? azurerm_public_ip.vm.*.ip_address : azurerm_network_interface.vm.*.private_ip_address,
-          azurerm_network_interface.vm.*.private_ip_address,
-        )
+}
+
+data "null_data_source" "node_ips" {
+  count = length(azurerm_virtual_machine.this.*.id)
+  inputs = {
+    id = element(azurerm_virtual_machine.this.*.id, count.index)
+    ip = element(coalescelist(azurerm_public_ip.vm.*.ip_address, azurerm_network_interface.vm.*.private_ip_address), count.index)
```

Comment:
Mentioned in slack, and in a comment of a similar aws tf change, but ranchhand tf expect a string with `public:private`. We should either deprecate that and only support private clusters, or still form it properly, since giving rke just the public ips I think doesn't work.

Reply:
Done :+1: